### ISSUE TITLE
sort-comp requires functions to be ordered alphabetically

### ISSUE THREAD

**Issue by NeroVanbiervliet** [yannickcr/eslint-plugin-react#2488]
[sort-comp](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md) fails when functions are not sorted alphabetically. 

**Fails**
```js
constructor(props) { ... }

bFunction() { ... }
aFunction() { ... }

render() { ... }

```

**Passes**
```js
constructor(props) { ... }

aFunction() { ... }
bFunction() { ... }

render() { ... }

```

Can this be disabled?

**Comment by golopot** [comment]
This is working as intended. If you want to disable the rule, you can set `{"rules": {"react/sort-comp": "off"}}` in your `.eslintrc.json`.

**Comment by NeroVanbiervliet** [comment]
I do like the rule, as it enforces the render method to be last, for example. But the alphabetically sorting of custom methods is not mentioned in the [docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md). Can I disable the alphabetical part of the rule separately?

**Comment by ljharb** [comment]
The rule is a "sort" rule that happens to have groups; without alpha sorting, there'd be nothing but grouping, and then the rule name would be inaccurate. Is it that you want *no* sorting of your custom methods? Why?

**Comment by NeroVanbiervliet** [comment]
Thanks for the reply! I don't think the rule name would be inaccurate: the *groups* are still sorted in a defined order. I just don't want sorting *inside* a group. Why? Because I like my functions to be sorted from high level functions at the top to functions with the most implementation details at the bottom.

**Comment by ljharb** [comment]
I suppose that seems reasonable - an option to disable alpha sorting.

**Comment by tanmoyopenroot** [comment]
I am not sure but on looking at the [line](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/sort-comp.js#L355) it looks like if two methods like `bFunction` and `aFunction` will have same `refIndex` so it should not complain to sort the custom methods.

Please correct me if i am wrong.

**Comment by NeroVanbiervliet** [comment]
Well spotted, seems like a mistake on my side. `bFunction` was declared as a [class property](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties) using an arrow function:
```
bFunction = () => {
...
}
```
When I change both to normal functions, the linter is happy. I'm closing this issue.

**Comment by ljharb** [comment]
Definitely you never want functions in class fields :-) glad you fixed it.
